stock_picking_invoice_link/models/stock_move.py

```diff
@@ -30,7 +31,14 @@ def write(self, vals):
         if "product_uom_qty" in vals and not self.env.context.get(
             "bypass_stock_move_update_restriction"
         ):
-            for move in self:
+            for move in self.filtered(
+                lambda x: float_compare(
+                    x.product_uom_qty,
+                    vals.get("product_uom_qty"),
```

Comment:
Can you put a fallback value of 0.0. Just in case, as it it possible that you have the key in vals but with None as value.

Reply:
Done!